logrusx/helper.go

```diff
@@ -98,6 +99,62 @@ func (l *Logger) WithRequest(r *http.Request) *Logger {
 	return ll
 }
 
+func (l *Logger) Logf(level logrus.Level, format string, args ...interface{}) {
+	if !l.leakSensitive {
+		var myArgs []interface{}
+		for _, arg := range args {
+			urlArg, ok := arg.(*url.URL)
+			if ok {
+				urlCopy, _ := url.Parse(urlArg.String())
```

Comment:
```suggestion
				urlCopy := *urlArg
```
The only field that will not be properly copied by this is the userinfo, and I suggest dropping that anyways.